components/scream/src/physics/shoc/shoc_main_impl.hpp

```diff
@@ -378,6 +645,44 @@ Int Functions<S,D>::shoc_main(
     shoc_output.pblh(i) = pblh_s;
   });
   Kokkos::fence();
+#else
+  const auto u_wind_s   = Kokkos::subview(shoc_input_output.horiz_wind, Kokkos::ALL(), 0, Kokkos::ALL());
+  const auto v_wind_s   = Kokkos::subview(shoc_input_output.horiz_wind, Kokkos::ALL(), 1, Kokkos::ALL());
+
+  EKAT_REQUIRE_MSG(workspace_mgr.get_max_slots() == shcol, "Cannot use workspace sharing with short kernels");
+
+  shoc_main_internal(shcol, nlev, nlevi, npbl, nadv, num_qtracers, dtime,
+    shoc_input.dx, shoc_input.dy, shoc_input.zt_grid, shoc_input.zi_grid, // Input
+    shoc_input.pres, shoc_input.presi, shoc_input.pdel, shoc_input.thv, shoc_input.w_field, // Input
+    shoc_input.wthl_sfc, shoc_input.wqw_sfc, shoc_input.uw_sfc, shoc_input.vw_sfc, // Input
+    shoc_input.wtracer_sfc, shoc_input.inv_exner, shoc_input.phis, // Input
+    workspace_mgr, // Workspace Manager
+    shoc_input_output.host_dse, shoc_input_output.tke, shoc_input_output.thetal, shoc_input_output.qw, u_wind_s, v_wind_s, // Input/Output
+    shoc_input_output.wthv_sec, shoc_input_output.qtracers, shoc_input_output.tk, shoc_input_output.shoc_cldfrac, // Input/Output
+    shoc_input_output.shoc_ql, // Input/Output
+    shoc_output.pblh, shoc_output.shoc_ql2, // Output
+    shoc_history_output.shoc_mix, shoc_history_output.w_sec, shoc_history_output.thl_sec, shoc_history_output.qw_sec, shoc_history_output.qwthl_sec, // Diagnostic Output Variables
+    shoc_history_output.wthl_sec, shoc_history_output.wqw_sec, shoc_history_output.wtke_sec, shoc_history_output.uw_sec, shoc_history_output.vw_sec, // Diagnostic Output Variables
+    shoc_history_output.w3, shoc_history_output.wqls_sec, shoc_history_output.brunt, shoc_history_output.isotropy, // Diagnostic Output Variables
+    shoc_temporaries.se_b,
```

Comment:
I followed the existing pattern of unpacking the struct before calling shoc_main_internal even though this function has a crazy number of arguments.